tinygraph/tinygraph.h

```diff
@@ -196,6 +196,26 @@ TINYGRAPH_API
 TINYGRAPH_WARN_UNUSED
 uint32_t tinygraph_size_in_bytes(tinygraph_const_s graph);
 
+/**
+ * Reorders graph nodes based on their spatial embedding
+ * in the `n` sized `lngs` and `lats` parallel arrays:
+ *
+ * - `nodes[i]` the node i's id (can be the identity)
+ * - `lngs[i]` the longitude fixed-point for node i
+ * - `lats[i]` the latitude fixed-point for node i
+ *
+ * The caller is responsible for providing the n sized
+ * `nodes` array which will get spatially sorted inplace.
+ *
+ * It is recommended to reorder graph nodes spatially
+ * for better memory locality and reduced cache misses.
+ */
+TINYGRAPH_API
+void tinygraph_reorder(
+    uint32_t* nodes,
+    const uint32_t* lngs,
+    const uint32_t* lats,
+    uint32_t n);
```

Comment:
New library API function

Reply:
The issue with using u32 for longitudes and latitudes is assuming a graph with max nodes (2^32)
1. with u32 the total space needed will be 48GB
2. with u16 for lngs and lats the total space needed will be 32GB

u32 gives precision <1cm and u16 gives precision <10m, for spatially sorting u16 is therefore good enough.

I'm still moving ahead with u32 for ease of API usage and because users will want to store u32 locations anyway.